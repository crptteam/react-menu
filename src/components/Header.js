import React, { PureComponent } from 'react';
import PropTypes from 'prop-types';

import HeaderWrapper from '../styled/HeaderWrapper';


export default class Header extends PureComponent {
  displayName = 'Header';

  static propTypes = {
    isOpen: PropTypes.bool.isRequired,
    content: PropTypes.func.isRequired,
  };

  render() {
    const {
      isOpen,
      content: HeaderContent,
    } = this.props;
    return (
      <HeaderWrapper>
        <HeaderContent isOpen={isOpen} />
      </HeaderWrapper>
    );
  }
}
