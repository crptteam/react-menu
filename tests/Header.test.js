import React from 'react';
import { shallow } from 'enzyme';

import Header from '../src/components/Header';

describe('Header', () => {
  it('should render without problem', () => {
    const component = shallow(<Header isFull content={() => (<div />)} />);
    expect(component).toMatchSnapshot();
  });
});

describe('HeaderContent', () => {
  it('should render with correct isFull', () => {
    const headerContent = jest.fn();
    const component = shallow(<Header isFull content={headerContent} />);

    expect(
      component
        .find('WithTheme(HeaderWrapper)')
        .children()
        .at(0)
        .prop('isFull'),
    ).toEqual(true);

    component.setProps({ isFull: false });
    expect(
      component
        .find('WithTheme(HeaderWrapper)')
        .children()
        .at(0)
        .prop('isFull'),
    ).toEqual(false);
  });
});
