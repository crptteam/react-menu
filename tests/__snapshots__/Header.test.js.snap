// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header should render without problem 1`] = `
<WithTheme(HeaderWrapper)>
  <content
    isFull={true}
  />
</WithTheme(HeaderWrapper)>
`;
